rendermodules/intensity_correction/apply_multiplicative_correction.py

```diff
@@ -70,50 +72,82 @@ def getImage(ts):
     ts: renderapi.tilespec.TileSpec
         tilespec to get images from
         (presently assumes this is a tiff image whose URL can be read with tifffile)
-
+    channel: str
+        channel name to get image of, default=None which will default to the non channel image pyramid
     Returns
     =======
     numpy.array
         2d numpy array of this image
     """
-    d = ts.to_dict()
-    mml = ts.ip[0]
+    if channel is None:
+        mml = ts.ip[0]
+    else:
+        if ts.channels is None:
+            raise RenderModuleException('No channels exist for this tilespec (ts.tileId={}'.format(ts.tileId))
+        else:
+            try:
+                chan = next(ch for ch in ts.channels if ch.name==channel)
+                mml = chan.ip[0]
+            except StopIteration as e:
+                raise RenderModuleException('Tilespec {} does not contain channel {}'.format(ts.tileId,channel))
+
     url = urllib.parse.unquote(urllib.parse.urlparse(
         str(mml.imageUrl)).path)
     img0 = tifffile.imread(url)
     (N, M) = img0.shape
     return N, M, img0
 
+def write_image(dirout, orig_imageurl, Res, stackname, z):
+    [head, tail] = os.path.split(orig_imageurl)
+    outImage = os.path.join("%s" % dirout, "%s_%04d_%s" %
+                            (stackname, z, tail))
+    tifffile.imsave(outImage, Res)
+    return outImage
 
-def process_tile(C, dirout, stackname, clip,scale_factor,clip_min,clip_max,input_ts):
+def process_tile_multichan(C, corr_dict, dirout, stackname, clip, scale_factor, clip_min, clip_max, input_ts):
```

Comment:
can you make this optional keyword argument?

Reply:
I assumed I couldn't, because of the partial, but forgot that partial can take a named argument. I'll adjust.

Reply:
Making it optional is a bit messy with the call via partial().  It can't be the last argument -- that's reserved for the tilespec. I'm not sure what the easy solution is without reworking the execution.

Reply:
partial works fine with keyword arguments that come after... so you if you have f(a,b,c,d=None) and you do myp= partial(f, 1,2,d=5). then myp(4) is f(a=1, b=2, c=4,d=5)

Reply:
I thought I had tried this already, but must have forgotten to change the order of arguments passed in. Swapped back to just process_tile(...corr_dict=None), and it seems to work.